src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,278 @@
+      i18n.unregisterTranslations();
+    });
+
+    describe('getTranslationsForLocale', function () {
+
+      it('en translations are registered' , function () {
+        const locale = 'en';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!',
+          'test_plugin_2-XXXXXX': 'This is XXXXXX string',
+          'test_plugin_2-YYYY_PPPP': 'This is YYYY_PPPP string',
+          'test_plugin_2-FFFFFFFFFFFF': 'This is FFFFFFFFFFFF string',
+          'test_plugin_2-ZZZ': 'This is ZZZ string'
+        };
+        checkTranslations(locale, expectedTranslationJson);
+        return;
+      });
+
+      it('de translations are registered' , function () {
+        const locale = 'de';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        checkTranslations(locale, expectedTranslationJson);
+        return;
+      });
+
+      it('Translations are registered with non unique key' , function () {
+        const pluginName = 'test_plugin_2';
+        const pluginTranslationPath = path.join(FIXTURES, 'translations', pluginName);
+        const translationFiles = [ path.join(pluginTranslationPath, 'de.json') ];
+        translationFiles.forEach(i18n.registerTranslations);
+        const locale = 'de';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS! I am regsitered afterwards!',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        checkTranslations(locale, expectedTranslationJson);
+        return;
+      });
+
+    });
+
+    describe('getRegisteredTranslationLocales', function () {
+
+      it('Translation locales are registered', function () {
+        const expectedLocales = ['en', 'de'];
+        checkRegisteredLocales(expectedLocales);
+        return;
+      });
+
+    });
+
+  });
+
+  describe('registerTranslations', function () {
+
+    it('Empty filename', function () {
+      try {
+        i18n.registerTranslations('');
+      } catch (err) {
+        expect(err).to.eql(new Error());
+      }
+      return;
+    });
+
+    it('Filename no extension', function () {
+      try {
+        i18n.registerTranslations('file1');
+      } catch (err) {
+        expect(err).to.eql(new Error());
+      }
+      return;
+    });
+
+    it('Filename non JSON extension', function () {
+      try {
+        i18n.registerTranslations('file1.txt');
+      } catch (err) {
+        expect(err).to.eql(new Error());
+      }
+      return;
+    });
```

Comment:
Scratch my above comment; must have gone to sleep! :-) Updated it in the following [commit](https://github.com/elastic/kibana/pull/7545/commits/cc380ba95eac3a8b2adf04f761c5c953bfa7bab1).